### PR TITLE
Feature/type filtering in dialog

### PR DESCRIPTION
**Pull request by hmaryam84**
#### **Summary**  
This pull request introduces a new filter option to display only **PRO** activities, removes the **SOLO** activity type, and updates the associated logic and tests to align with these changes. Minor code refactoring for formatting is also included.  

---

#### **Key Features**  

1. **"Only See PRO Activities" Checkbox**  
   - Added a checkbox in the filter dialog for users to filter activities exclusively by the **PRO** type.  
   - Updated the filtering logic to accommodate this new option.  

2. **Remove SOLO Activity Type**  
   - The **SOLO** activity type has been removed from the codebase.  
   - Updated tests and filtering mechanisms to exclude references to **SOLO** activities.  

3. **Test**  
   - Added and improved tests to verify the "Only See PRO Activities" filter functionality.  
   - Ensured existing tests work correctly with the removal of the SOLO type.  

---
#### **Testing**  
- **Unit Tests:**  
  - Verified the "Only See PRO Activities" checkbox functionality.  
  - Ensured filtering logic adapts correctly with the absence of the SOLO type.

### CODE REVIEW

File: app/src/main/java/com/android/sample/ui/dialogs/FilterDialog.kt [hunk 1]
```diff
@@ -28,13 +28,19 @@ import com.google.firebase.Timestamp
...
+            schedule: Timestamp?,
+            duration: String?,
+            onlyPRO: Boolean?) -> Unit,
 ) {
   var maxPrice by remember { mutableStateOf(DEFAULT_MAX_PRICE) }
   var availablePlaces by remember { mutableStateOf<Int?>(null) }
   var minDate by remember { mutableStateOf<String?>(null) }
   var minDateTimestamp by remember { mutableStateOf<Timestamp?>(null) }
   var duration by remember { mutableStateOf<String?>(null) }
+  var onlyPRO by remember { mutableStateOf<Boolean?>(false) }
```
Comment (Yassyass181104) [inline app/src/main/java/com/android/sample/ui/dialogs/FilterDialog.kt:43]: There is no need to specify the type but it works, nothing to say then

**Review by Yassyass181104 — APPROVED** [review]
You implemented the possibility to filter activities to see only the pro ones which is a good feature. This allows the user to have the certainty to deal with a serious organization and not with a normal user. You also deleted the segmented button that showed comfusing categories. Good job !

File: app/src/main/java/com/android/sample/ui/dialogs/FilterDialog.kt [hunk 2]
```diff
@@ -103,12 +109,31 @@ fun FilterDialog(
...
+                            onCheckedChange = { onlyPRO = it },
+                            colors =
+                                CheckboxDefaults.colors(
+                                    checkedColor = MaterialTheme.colors.primary),
+                        )
+                        Text(
+                            "Only see PRO activities",
+                        )
+                      }
+                }
```
Comment (aminebenji04) [inline app/src/main/java/com/android/sample/ui/dialogs/FilterDialog.kt:128]: LGTM, you just added the PRO filter in the already available filter list

**Review by aminebenji04 — COMMENTED** [review]
Your PR is very understandable and targets a very specofoc problem, there is no inconsistency in your code, no magic values, and everything is clean. The tests are good and this is proved by your line coverage, goof job!

**Review by aminebenji04 — APPROVED** [review]
Your PR is very understandable and targets a very specofoc problem, there is no inconsistency in your code, no magic values, and everything is clean. The tests are good and this is proved by your line coverage, goof job!

**Pull request merged**